### ISSUE TITLE
Project corrupted when creting a parameter and changing dimension

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/MoBi#411]
This one almost got me into psychiatry until I figured out what is the problem...

1. Open MoBi. Create new project (Amount-based)
2. Open the Spatial Structures-BB and navigate to "Organism". Select the tab "Parameters"
3. Click on "Add Parameter".
4. Type in any name (`myParam`)
5. Select dimension "Dimensionless"
6. Select dimension "Time"
7. Click "OK.
8. Save project and close MoBi.
9. Open the project

```
Value cannot be null.
Parameter name: key

at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at OSPSuite.Utility.Collections.Cache`2.Add(TKey key, TValue value)
   at OSPSuite.Core.Domain.UnitSystem.Dimension.AddUnit(Unit unit)
   at OSPSuite.Serializer.Xml.XmlEnumerationNodeMapper`3.Deserialize(TObject objectToDeserialize, XElement element, TContext context)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.TypedDeserialize(TObject objectToDeserialize, XElement outputToDeserialize, TContext context)
   at OSPSuite.Core.Serialization.Xml.DimensionXmlSerializer.TypedDeserialize(Dimension dimension, XElement element, SerializationContext serializationContext)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.Deserialize(XElement outputToDeserialize, TContext context)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.Deserialize[T](XElement outputToDeserialize, TContext context)
   at OSPSuite.Serializer.Xml.XmlEnumerationNodeMapper`3.valueFromChildNode(XElement childElement, TContext context)
   at OSPSuite.Serializer.Xml.XmlEnumerationNodeMapper`3.Deserialize(TObject objectToDeserialize, XElement element, TContext context)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.TypedDeserialize(TObject objectToDeserialize, XElement outputToDeserialize, TContext context)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.Deserialize(Object objectToDeserialize, XElement outputToDeserialize, TContext context)
   at MoBi.Core.Serialization.Xml.Serializer.MergedDimensionFactorySerializer.TypedDeserialize(DimensionFactory objectToDeserialize, XElement outputToDeserialize, SerializationContext serializationContext)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.Deserialize(XElement outputToDeserialize, TContext context)
   at OSPSuite.Serializer.Xml.XmlStaticMapper`2.Deserialize(TObject objectToDeserialize, XElement element, TContext context)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.TypedDeserialize(TObject objectToDeserialize, XElement outputToDeserialize, TContext context)
   at MoBi.Core.Serialization.Xml.Serializer.MoBiProjectXmlSerializer.TypedDeserialize(MoBiProject project, XElement outputToDeserialize, SerializationContext serializationContext)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.Deserialize(XElement outputToDeserialize, TContext context)
   at MoBi.Core.Serialization.Xml.Services.XmlSerializationService.deserialize(XElement element, IMoBiProject project, Int32 version, Type type, SerializationContext parentSerializationContext)
   at MoBi.Core.Serialization.Xml.Services.XmlSerializationService.Deserialize[T](Byte[] compressedBytes, IMoBiProject project, SerializationContext serializationContext)
   at MoBi.Core.Serialization.ORM.Mappers.ProjectMetaDataToProjectMapper.deserializeContent[T](MetaDataContent content, SerializationContext serializationContext)
   at MoBi.Core.Serialization.ORM.Mappers.ProjectMetaDataToProjectMapper.MapFrom(ProjectMetaData projectMetaData)
   at MoBi.Core.Serialization.ORM.ProjectPersistor.projectFrom(ProjectMetaData projectMetaData)
   at MoBi.Core.Serialization.ORM.ProjectPersistor.Load(IMoBiContext context)
   at MoBi.Core.Serialization.ORM.ContextPersistor.Load(IMoBiContext context, String projectFullPath)
   at MoBi.Presentation.Tasks.SerializationTask.<>c__DisplayClass10_0.<LoadProject>b__0()
   at OSPSuite.Presentation.Services.HeavyWorkManager.doWork()
   at OSPSuite.Presentation.Services.HeavyWorkManager.<.ctor>b__9_1(Object o, DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.OnDoWork(DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)
```

**Comment by PavelBal** [comment]
OK, this happens **every time** I change the dimension in the "Create parameter" window...

**Comment by UlrichSi** [comment]
Naughty you! ;-)

I think we have come a long way finding such bugs that only show up in the "Create" window, but not on later editing. Still haven't found all of them. :-(

**Comment by msevestre** [comment]
This has been fixed with #404.
